CalibTracker/SiStripChannelGain/plugins/DeDxDiscriminatorLearner.cc

```diff
@@ -33,6 +33,7 @@ using namespace edm;
 
 DeDxDiscriminatorLearner::DeDxDiscriminatorLearner(const edm::ParameterSet& iConfig)
     : ConditionDBWriter<PhysicsTools::Calibration::HistogramD3D>(iConfig) {
+  usesResource();
```

Comment:
I see now that my comment https://github.com/cms-sw/cmssw/pull/40754#issuecomment-1431675916 was confusing, apologies. I should have written that, especially now that the bcase class `CondDBWriter<T>` calls the `usesResource(cond::service::PoolDBOutputService::kSharedResource)`, the deriving classes do not need to call `usesResource()`, unless they use some additional shared resource. I would naively expect that most of the deriving classes do not need the call at all, and the ones that use `TFileService` would need `usesResource(TFileService::kSharedResource)`.